apps/condo/domains/miniapp/schema/B2BAppAccessRightSet.test.js

```diff
@@ -344,14 +346,120 @@ describe('B2BAppAccessRightSet', () => {
+            const [app] = await createTestB2BApp(support)
+            await createTestB2BAppAccessRightSet(support, app)
+            for (let i = 0; i < ACCESS_RIGHT_SET_MAX_ITEMS_SCOPED_TYPE; i++) {
+                await createTestB2BAppAccessRightSet(support, app, { type: 'SCOPED' })
+            }
+
+            await expectToThrowGQLError(async () => {
+                await createTestB2BAppAccessRightSet(support, app, { type: 'SCOPED' })
+            }, {
+                code: 'BAD_USER_INPUT',
+                type: 'ACCESS_RIGHT_SET_TOO_MANY_OF_TYPE',
+                message: `Too many items of type "SCOPED". Maximum is "${ACCESS_RIGHT_SET_MAX_ITEMS_SCOPED_TYPE}"`,
+            }, 'obj')
+        })
+
+        test('Cannot create or update set with type "SCOPED" if type "GLOBAL" doesn\'t exists', async () => {
+            const [app] = await createTestB2BApp(support)
+
+            await expectToThrowGQLError(async () => {
+                await createTestB2BAppAccessRightSet(support, app, { type: 'SCOPED' })
+            }, {
+                code: 'BAD_USER_INPUT',
+                type: 'ACCESS_RIGHT_SET_GLOBAL_RIGHT_SET_REQUIRED',
+                message: 'You need to have "GLOBAL" type right before other types',
+            }, 'obj')
+        })
+
+        test('Can not create or update B2BAppAccessRightSet of type "SCOPED" with more permissions, than set of type "GLOBAL" on same app', async () => {
+            const [app] = await createTestB2BApp(support)
+            await createTestB2BAppAccessRightSet(support, app, { canReadOrganizations: true })
+
+            await expectToThrowGQLError(async () => {
+                await createTestB2BAppAccessRightSet(support, app, { type: 'SCOPED', canReadMeters: true })
+            }, {
+                code: 'BAD_USER_INPUT',
+                type: 'ACCESS_RIGHT_SET_TOO_MANY_PERMISSIONS',
+                message: 'You trying to give to right set more permissions, than "GLOBAL" right set has',
+            }, 'obj')
+
+            const [tokenRightSet] = await createTestB2BAppAccessRightSet(support, app, { type: 'SCOPED', canReadOrganizations: true })
+
+            await expectToThrowGQLError(async () => {
+                await updateTestB2BAppAccessRightSet(support, tokenRightSet.id, { canReadMeters: true })
+            }, {
+                code: 'BAD_USER_INPUT',
+                type: 'ACCESS_RIGHT_SET_TOO_MANY_PERMISSIONS',
+                message: 'You trying to give to right set more permissions, than "GLOBAL" right set has',
+            }, 'obj')
+        })
+
+        test('Removing permissions on "GLOBAL" B2BAppAccessRightSet leads to removing these permissions from entities with type "SCOPED"', async () => {
+            const [app] = await createTestB2BApp(support)
+            const [globalRightsSet] = await createTestB2BAppAccessRightSet(support, app, { canReadOrganizations: true })
+
+            for (let i = 0; i < 3; i++) {
```

Comment:
<3

Reply:
It's actually not testing maximum count of entities, just constant to create some right sets for testing, not related to anything.
Made this number random between 1 and maximum scoped right sets count